### PR TITLE
Fix NU1105 errors in HttpAbstractions.sln

### PR DESCRIPTION
**Pull request by JamesNK**
Workaround NU1105 errors by adding referenced projects to solution

### CODE REVIEW

**Comment by JamesNK** [comment]
FYI the routing projects were in the root of the solution, I'm guessing because of sln GUIDs and merging from other branches.

This is where I moved everything. Let me know if you want the folder structure changed:

image

**Comment by natemcmaster** [comment]
I might have put the new projects you've added into the dependencies/ solution folder, but don't care too much either way. Thoughts @Tratcher ?

**Review by natemcmaster — APPROVED** [review]

File: src/Http/HttpAbstractions.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26124.0
+# Visual Studio Version 16
```
Comment (natemcmaster) [inline src/Http/HttpAbstractions.sln]: Does this still open in VS 2017? We haven't started requiring VS 2019 (yet)

Comment (JamesNK) [reply src/Http/HttpAbstractions.sln]: It does

**Comment by JamesNK** [comment]
Makes sense. I'll move all the dependencies to dependencies 😁

**Comment by Tratcher** [comment]
HttpAbstractions doesn't depend on Hosting or Servers, it's the other way around. What was the full error that implied this?

**Comment by Tratcher** [comment]
Or did routing bring these in?

**Comment by JamesNK** [comment]
I looked to Middleware.sln for what it does. Updated folder structure:

image

**Comment by JamesNK** [comment]
> Or did routing bring these in?

The sample and functional tests websites brought them in.

image

**Comment by Tratcher** [comment]
Ah, that would have been a big circular repo dependency nightmare before.

**Comment by rynowak** [comment]
IMO we should consider a different factoring for this. We have a plan to enable functional test on the shared framework - where layering is not a concern.

**Comment by Tratcher** [comment]
> IMO we should consider a different factoring for this. We have a plan to enable functional test on the shared framework - where layering is not a concern.

What kind of different factoring?

**Comment by rynowak** [comment]
> What kind of different factoring?

I just generally chafe at putting a million things into a "sample" that isn't a sample of anything. We created a "sample" project in MVC that's virtually empty and stays that way, because it's there for you to have a canvas to test your changes in the repo.

**Comment by JamesNK** [comment]
At the moment the sandbox contains prototypes for scenarios that happen to be checked in. AuthZ middleware can be deleted now that it is done for real, the middleware example could be deleted once healthchecks is endpointer-ized, etc.

Can clean it up later.

**Pull request merged**